### PR TITLE
Benchmark script

### PR DESCRIPTION
**Pull request by rengolin**
Trying to create a script that can help external users to build and benchmark our project. Preparing for the AWS runs before we book the instances and start paying for it, so testing locally for now.

Some fixes for OpenMP arguments that were out of sync between `tpp-run` and `xsmm_dnn_mlp` which made a difference on a newer Ubuntu.

@alheinecke

### CODE REVIEW

**Comment by rengolin** [comment]
The benchmark script looks ok, works on my Arch box, but not on my Ubuntu server box, which I'm trying to get in sync with AWS. the OpenMP problem is what's creating all this. I'll see if I can get an older Ubuntu if the problem persists.

**Comment by rengolin** [comment]
> The benchmark script looks ok, works on my Arch box, but not on my Ubuntu server box, which I'm trying to get in sync with AWS. the OpenMP problem is what's creating all this. I'll see if I can get an older Ubuntu if the problem persists.

Ubuntu Server 23.10 is completely broken, I got 22.04 LTS and it seems better. Still some issues with symbols for some benchmarks, not all, but almost there.

**Comment by rengolin** [comment]
> Ubuntu Server 23.10 is completely broken, I got 22.04 LTS and it seems better. Still some issues with symbols for some benchmarks, not all, but almost there.

all good, got a full run on my ubuntu server similar to aws

File: scripts/benchmarks/build_and_run.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# and run all benchmarks. This should work in the same way as our local tests
+# and reproduce our numbers on local/cloud machines.
+
+# Include common utils
+SCRIPT_DIR=$(realpath $(dirname $0)/..)
+source ${SCRIPT_DIR}/ci/common.sh
+
+# Install packages needed
+if [ "$(is_linux_distro Ubuntu)" == "YES" ]; then
+  sudo apt update && \
```
Comment (alheinecke) [inline scripts/benchmarks/build_and_run.sh:13]: I suggest using conda, this is much more portable.

Comment (rengolin) [reply scripts/benchmarks/build_and_run.sh]: Conda faffles with `.bashrc` and needs reloading stuff. I found the `dnf` install line for "Amazon Linux" that just worked on different instances on AWS.

**Pull request closed without merging**